library/proc_macro/src/bridge/client.rs

```diff
@@ -506,3 +510,26 @@ impl ProcMacro {
         ProcMacro::Bang { name, client: Client::expand1(expand) }
     }
 }
+
+#[const_trait]
+pub trait ExpandCustomDerive<Discriminant> {
+    fn into_fn(self) -> impl Fn(crate::TokenStream) -> crate::TokenStream + Copy;
+}
+
+impl<F> const ExpandCustomDerive<()> for F
+where
+    F: Fn(crate::TokenStream) -> crate::TokenStream + Copy,
+{
+    fn into_fn(self) -> impl Fn(crate::TokenStream) -> crate::TokenStream + Copy {
+        self
+    }
+}
+
+impl<F> const ExpandCustomDerive<crate::DeriveExpansionOptions> for F
+where
+    F: Fn(crate::TokenStream, crate::DeriveExpansionOptions) -> crate::TokenStream + Copy,
+{
+    fn into_fn(self) -> impl Fn(crate::TokenStream) -> crate::TokenStream + Copy {
+        move |input| self(input, Default::default())
```

Comment:
This feels a bit weird and it demonstrates that `DeriveExpansionOptions` is a “sham”. I fear that the way it's wired right now, this impl isn't extensible. Consider the hypothetical in which we *do* store something meaningful in `DeriveExpansionOptions`, we wouldn't be able to retrieve it, well with `FreeFunctions`we could I guess. Maybe this is a non-issue, I'm just rambling at this point ^^'